tests/test_restriction.cpp

```diff
@@ -0,0 +1,256 @@
+#include <gtest/gtest.h>
+#include "gmgpolar.h"
+class test_restriction : public ::testing::TestWithParam<int>
+{
+protected:
+    void SetUp() override
+    {
+        //initialize default parameters.
+        gyro::init_params();
+        gyro::icntl[Param::verbose]        = 0;
+        gyro::icntl[Param::debug]          = 0;
+        gyro::icntl[Param::extrapolation]  = 0;
+        gyro::icntl[Param::DirBC_Interior] = 1;
```

Comment:
I think we should test with and without Dirichlet conditions

Reply:
I can do this, but should the boundary conditions affect the operator ?
I thought that if we prolongate a solution u, then the information on boundary conditions should be stored in the stiffness matrix  and right-hand side not neccessarily in the prolongation and restriction operator.

Reply:
No they shouldnt. And it should be ensured that we have the same operator.